### PR TITLE
Add a few new `ItemModelProvider` methods!

### PR DESCRIPTION
**Pull request by ZestyBlaze**
Since `ItemModelProvider` only provides `basicItem` which does exactly what it says on the tin, when having to do other items that require something a bit different, such as handheld items or any item that supports a template from Minecraft or your own mod, it requires the developer to write a custom method just to handle these item models or hand write them
This PR adds two new methods at the moment, `handheldItem` and `spawnEggItem`, however this PR will be marked as draft for a while awaiting community input for more devs to suggest other useful methods they might like to see \:)

### CODE REVIEW

**Comment by ZestyBlaze** [comment]
New method added, PR is ready for review

File: src/main/java/net/neoforged/neoforge/client/model/generators/ItemModelProvider.java [hunk 1]
```diff
@@ -31,6 +32,33 @@ public ItemModelBuilder basicItem(ResourceLocation item) {
...
+    public ItemModelBuilder spawnEggItem(ResourceLocation item) {
+        return getBuilder(item.toString())
+                .parent(new ModelFile.UncheckedModelFile("item/template_spawn_egg"));
+    }
+
+    public ItemModelBuilder simpleBlockItem(Block block) {
+        return simpleBlockItem(Objects.requireNonNull(BuiltInRegistries.BLOCK.getKey(block)));
+    }
+
+    public ItemModelBuilder simpleBlockItem(ResourceLocation item) {
```
Comment (XFactHD) [inline src/main/java/net/neoforged/neoforge/client/model/generators/ItemModelProvider.java]:
```suggestion
    public ItemModelBuilder simpleBlockItem(ResourceLocation block) {
```

**Review by XFactHD — CHANGES_REQUESTED** [review]
This has been sitting long enough, if anyone comes up with further ideas, then a new PR is always an option.
One minor nitpick, then it's good to go.

**Comment by ZestyBlaze** [comment]
@XFactHD change made to resolve your review

File: src/main/java/net/neoforged/neoforge/client/model/generators/ItemModelProvider.java [hunk 2]
```diff
@@ -31,6 +32,33 @@ public ItemModelBuilder basicItem(ResourceLocation item) {
...
+        return getBuilder(item.toString())
+                .parent(new ModelFile.UncheckedModelFile("item/template_spawn_egg"));
+    }
+
+    public ItemModelBuilder simpleBlockItem(Block block) {
+        return simpleBlockItem(Objects.requireNonNull(BuiltInRegistries.BLOCK.getKey(block)));
+    }
+
+    public ItemModelBuilder simpleBlockItem(ResourceLocation block) {
+        return withExistingParent(item.toString(), ResourceLocation.fromNamespaceAndPath(item.getNamespace(), "block/" + item.getPath()));
```
Comment (XFactHD) [inline src/main/java/net/neoforged/neoforge/client/model/generators/ItemModelProvider.java]: The uses of the parameter need to be changed as well.
```suggestion
        return withExistingParent(block.toString(), ResourceLocation.fromNamespaceAndPath(block.getNamespace(), "block/" + block.getPath()));
```

**Review by XFactHD — CHANGES_REQUESTED** [review]

**Comment by ZestyBlaze** [comment]
Let's ignore how I totally forgot about that and just focus on the fact I've fixed it

**Review by XFactHD — CHANGES_REQUESTED** [review]
Please commit the generate files from the datagen test

**Review by XFactHD — APPROVED** [review]

**Pull request merged**